### PR TITLE
@ashfurrow => Add support for buyers premium - fixes #82

### LINKED ISSUE #82: Add Buyer's Premium Functionality
Some artworks in an auction will have a "buyer's premium" this will be noted underneath the BID button 
image

It will lead to a modal (martsy view) containing the relevant information explaining what a buyer's premium is and what that is for this particular sale (premiums change from sale to sale)

### PR DESCRIPTION
**Pull request by orta**
selfie-1

will inline

### CODE REVIEW

File: Artsy Tests/Sale+Extensions.m [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 + (Sale *)saleWithStart:(NSDate *)start end:(NSDate *)end
 {
     return [Sale modelFromDictionary:@{ @"startDate" : start, @"endDate" : end }];
 }
+
++ (Sale *)saleWithBuyersPremium
+{
+    return [Sale modelFromDictionary:@{ @"buyers_premium" : @{ @"id" : @"----" }}];
+}
+
```
Comment (orta) [inline Artsy Tests/Sale+Extensions.m]: seemed like a reasonable pattern

File: Artsy/Classes/Models/BuyersPremium.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#import "MTLModel.h"
+
+@interface BuyersPremium : MTLModel <MTLJSONSerializing>
+
+@property (nonatomic, copy, readonly) NSString *buyersPremiumID;
```
Comment (orta) [inline Artsy/Classes/Models/BuyersPremium.h]: we only really need to validate presence on the eigen side

File: Artsy/Classes/Utils/ARSwitchBoard.h [hunk 3]
```diff
@@ -62,6 +62,7 @@
 
 /// Load a path relative to the baseURL through the router
 - (UIViewController *)loadPath:(NSString *)path;
+- (UIViewController *)loadPath:(NSString *)path fair:(Fair *)fair;
```
Comment (orta) [inline Artsy/Classes/Utils/ARSwitchBoard.h]: was missing from when I added some more fair DI stuff

File: Artsy/Classes/View Controllers/ARArtworkViewController+ButtonActions.m [hunk 4]
```diff
@@ -20,7 +20,7 @@ - (void)tappedTileableImagePreview:(ARArtworkPreviewImageView *)sender
 {
     ARZoomArtworkImageViewController *zoomImgeVC = [[ARZoomArtworkImageViewController alloc] initWithImage:self.artwork.defaultImage];
     zoomImgeVC.suppressZoomViewCreation = (self.fair == nil);
-    [self.navigationController pushViewController:zoomImgeVC animated:YES];
+    [self.navigationController pushViewController:zoomImgeVC animated:self.shouldAnimate];
```
Comment (orta) [inline Artsy/Classes/View Controllers/ARArtworkViewController+ButtonActions.m]: this is always yes in app

**Comment by orta** [comment]
sidenote. I genuinely like the fact that adding a new view to our most complex view hierarchy can be super super easy.

**Comment by ashfurrow** [comment]
Nice and smooth. On the team now – feel free to merge without me :beers:

**Comment by orta** [comment]
rock

**Comment by orta** [comment]
This fail should be resolved in #158

**Pull request merged**